src/main/java/core/basesyntax/suppliers/FigureSupplier.java

```diff
@@ -0,0 +1,45 @@
+package core.basesyntax.suppliers;
+
+import core.basesyntax.Figure;
+import core.basesyntax.enums.Color;
+import core.basesyntax.enums.Figures;
+import core.basesyntax.figures.Circle;
+import core.basesyntax.figures.IsoscelesTrapezoid;
+import core.basesyntax.figures.Rectangle;
+import core.basesyntax.figures.RightTriangle;
+import core.basesyntax.figures.Square;
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_SIZE = 9;
+    private static final int DEFAULT_CIRCLE_RADIUS = 10;
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+    private final Random random = new Random();
+    private int figureIndex;
+    private Figures figures;
+
+    private int getRandomSize() {
+        return random.nextInt(MAX_SIZE) + 1;
+    }
+
+    public Figure getRandomFigure() {
+        figureIndex = random.nextInt(Figures.values().length);
+        figures = Figures.values()[figureIndex];
+        Color color = colorSupplier.getRandomColor();
+
+        return switch (figures) {
+            case Circle -> new Circle(getRandomSize(), color.name().toLowerCase());
```

Comment:
Move color.name().toLowerCase() to the variable
```suggestion
            case Circle -> new Circle(getRandomSize(), colorName);
```